Some tallies in dcp55 failed to index due to non-disjuctive projects again

[Failed tallies GL snippet: azul-tallies_fail-prod.fifo.2025-11-15.json](https://gitlab.azul.data.humancellatlas.org/ucsc/azul/-/snippets/15)

```
Traceback (most recent call last):
  File "/var/task/azul/indexer/index_controller.py", line 280, in aggregate
    self.index_queue_service.aggregate(tallies, retry=retry)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/indexer/index_queue_service.py", line 290, in aggregate
    self.index_service.aggregate(tally_by_entity)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/var/task/azul/indexer/index_service.py", line 489, in aggregate
    assert len(aggregate.sources) == 1, R(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: R('Entity has an invalid number of sources',
 CataloguedEntityReference(entity_type='samples', entity_id='a9920fc8-880e-4ebd-a1d5-b348d045f1fa', catalog='dcp55'),
 {
TDRSourceRef(id='0d69b7d1-bc70-4bd3-946a-db6d49042778', spec=TDRSourceSpec(type=<Type.bigquery: 'bigquery'>, domain=<Domain.gcp: 'gcp'>, subdomain='datarepo-b42d0841', name='hca_prod_9c20a245f2c043ae82c92232ec6b594f__20220212_dcp2_20251104_dcp55'), prefix=Prefix(common='', partition=0)),
 TDRSourceRef(id='1a449990-5cc7-4623-83a3-80440a0ab0bd', spec=TDRSourceSpec(type=<Type.bigquery: 'bigquery'>, domain=<Domain.gcp: 'gcp'>, subdomain='datarepo-03f25c54', name='hca_prod_4bcc16b57a4745bbb9c0be9d5336df2d__20240327_dcp2_20251104_dcp55'), prefix=Prefix(common='', partition=0)),
 TDRSourceRef(id='ce551a61-822f-4786-b878-c62576b0e006', spec=TDRSourceSpec(type=<Type.bigquery: 'bigquery'>, domain=<Domain.gcp: 'gcp'>, subdomain='datarepo-5cffc4da', name='hca_prod_2079bb2e676e4bbf8c68f9c6459edcbb__20240327_dcp2_20251104_dcp55'), prefix=Prefix(common='', partition=0))})
```

HCA Datasets that failed to index:

**CloudWatch Logs Insights**    
region: us-east-1    
log-group-names: /aws/lambda/azul-indexer-prod-aggregate_retry    
start-time: 2025-11-14T23:00:00.000Z    
end-time: 2025-11-15T23:00:00.000Z    
query-string:
  ```
  fields @timestamp, @message
| filter @message like 'attempts=9'
| sort @timestamp asc
| parse @message "entity_id='*'" as entity_id
| parse @message "name='*'" as dataset_name
| stats count(*) by dataset_name
| limit 10000
  ```
---
| dataset_name | count(*) |
| --- | --- |
| hca_prod_4bcc16b57a4745bbb9c0be9d5336df2d__20240327_dcp2_20251104_dcp55 | 8 |
| hca_prod_9c20a245f2c043ae82c92232ec6b594f__20220212_dcp2_20251104_dcp55 | 12 |
| hca_prod_2079bb2e676e4bbf8c68f9c6459edcbb__20240327_dcp2_20251104_dcp55 | 17 |
---


There are indeed 40 tallies, three executions seem to have a single logs regarding failures for two tallies, hence the three missing executions:

**CloudWatch Logs Insights**    
region: us-east-1    
log-group-names: /aws/lambda/azul-indexer-prod-aggregate_retry    
start-time: 2025-11-14T23:00:00.000Z    
end-time: 2025-11-15T23:00:00.000Z    
query-string:
  ```
  fields @timestamp, @message
| filter @message like 'attempts=9'
| filter @requestId like /e2f59684|969edacd|3490c834/
| sort @timestamp asc
| limit 10000
  ```
---
| @timestamp | @message |
| --- | --- |
| 2025-11-15 11:12:17.212 | [WARNING] 2025-11-15T11:12:17.211Z e2f59684-9ad3-5ab3-8301-f716f8ec3754 azul.indexer.index_controller Failed to aggregate tallies: [DocumentTally(entity=CataloguedEntityReference(entity_type=&#x27;samples&#x27;, entity_id=&#x27;1eb02140-6c37-493c-9dcc-c2f5fb8e4ae0&#x27;, catalog=&#x27;dcp55&#x27;), num_contributions=4, attempts=9), DocumentTally(entity=CataloguedEntityReference(entity_type=&#x27;samples&#x27;, entity_id=&#x27;c494198d-97df-42d7-9656-5bf9c2a04ac8&#x27;, catalog=&#x27;dcp55&#x27;), num_contributions=4, attempts=9)] Traceback (most recent call last):   File &quot;&#x2f;var&#x2f;task&#x2f;azul&#x2f;indexer&#x2f;index_controller.py&quot;, line 280, in aggregate     self.index_queue_service.aggregate(tallies, retry=retry)     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^   File &quot;&#x2f;var&#x2f;task&#x2f;azul&#x2f;indexer&#x2f;index_queue_service.py&quot;, line 290, in aggregate     self.index_service.aggregate(tally_by_entity)     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^   File &quot;&#x2f;var&#x2f;task&#x2f;azul&#x2f;indexer&#x2f;index_service.py&quot;, line 489, in aggregate     assert len(aggregate.sources) == 1, R(            ^^^^^^^^^^^^^^^^^^^^^^^^^^^ AssertionError: R(&#x27;Entity has an invalid number of sources&#x27;, CataloguedEntityReference(entity_type=&#x27;samples&#x27;, entity_id=&#x27;1eb02140-6c37-493c-9dcc-c2f5fb8e4ae0&#x27;, catalog=&#x27;dcp55&#x27;), {TDRSourceRef(id=&#x27;0d69b7d1-bc70-4bd3-946a-db6d49042778&#x27;, spec=TDRSourceSpec(type=&lt;Type.bigquery: &#x27;bigquery&#x27;&gt;, domain=&lt;Domain.gcp: &#x27;gcp&#x27;&gt;, subdomain=&#x27;datarepo-b42d0841&#x27;, name=&#x27;hca_prod_9c20a245f2c043ae82c92232ec6b594f__20220212_dcp2_20251104_dcp55&#x27;), prefix=Prefix(common=&#x27;&#x27;, partition=0)), TDRSourceRef(id=&#x27;ce551a61-822f-4786-b878-c62576b0e006&#x27;, spec=TDRSourceSpec(type=&lt;Type.bigquery: &#x27;bigquery&#x27;&gt;, domain=&lt;Domain.gcp: &#x27;gcp&#x27;&gt;, subdomain=&#x27;datarepo-5cffc4da&#x27;, name=&#x27;hca_prod_2079bb2e676e4bbf8c68f9c6459edcbb__20240327_dcp2_20251104_dcp55&#x27;), prefix=Prefix(common=&#x27;&#x27;, partition=0))}) |
| 2025-11-15 11:20:09.250 | [WARNING] 2025-11-15T11:20:09.249Z 969edacd-7918-507f-82c1-bb5f1d00687a azul.indexer.index_controller Failed to aggregate tallies: [DocumentTally(entity=CataloguedEntityReference(entity_type=&#x27;samples&#x27;, entity_id=&#x27;a75d5f39-acab-4b8f-88f2-64c7bdaa6752&#x27;, catalog=&#x27;dcp55&#x27;), num_contributions=13, attempts=9), DocumentTally(entity=CataloguedEntityReference(entity_type=&#x27;samples&#x27;, entity_id=&#x27;d276045f-d040-4f29-b316-167429bb6c16&#x27;, catalog=&#x27;dcp55&#x27;), num_contributions=9, attempts=9)] Traceback (most recent call last):   File &quot;&#x2f;var&#x2f;task&#x2f;azul&#x2f;indexer&#x2f;index_controller.py&quot;, line 280, in aggregate     self.index_queue_service.aggregate(tallies, retry=retry)     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^   File &quot;&#x2f;var&#x2f;task&#x2f;azul&#x2f;indexer&#x2f;index_queue_service.py&quot;, line 290, in aggregate     self.index_service.aggregate(tally_by_entity)     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^   File &quot;&#x2f;var&#x2f;task&#x2f;azul&#x2f;indexer&#x2f;index_service.py&quot;, line 489, in aggregate     assert len(aggregate.sources) == 1, R(            ^^^^^^^^^^^^^^^^^^^^^^^^^^^ AssertionError: R(&#x27;Entity has an invalid number of sources&#x27;, CataloguedEntityReference(entity_type=&#x27;samples&#x27;, entity_id=&#x27;a75d5f39-acab-4b8f-88f2-64c7bdaa6752&#x27;, catalog=&#x27;dcp55&#x27;), {TDRSourceRef(id=&#x27;ce551a61-822f-4786-b878-c62576b0e006&#x27;, spec=TDRSourceSpec(type=&lt;Type.bigquery: &#x27;bigquery&#x27;&gt;, domain=&lt;Domain.gcp: &#x27;gcp&#x27;&gt;, subdomain=&#x27;datarepo-5cffc4da&#x27;, name=&#x27;hca_prod_2079bb2e676e4bbf8c68f9c6459edcbb__20240327_dcp2_20251104_dcp55&#x27;), prefix=Prefix(common=&#x27;&#x27;, partition=0)), TDRSourceRef(id=&#x27;1a449990-5cc7-4623-83a3-80440a0ab0bd&#x27;, spec=TDRSourceSpec(type=&lt;Type.bigquery: &#x27;bigquery&#x27;&gt;, domain=&lt;Domain.gcp: &#x27;gcp&#x27;&gt;, subdomain=&#x27;datarepo-03f25c54&#x27;, name=&#x27;hca_prod_4bcc16b57a4745bbb9c0be9d5336df2d__20240327_dcp2_20251104_dcp55&#x27;), prefix=Prefix(common=&#x27;&#x27;, partition=0))}) |
| 2025-11-15 13:13:16.455 | [WARNING] 2025-11-15T13:13:16.454Z 3490c834-a9de-537f-80aa-a9a6d7459ff5 azul.indexer.index_controller Failed to aggregate tallies: [DocumentTally(entity=CataloguedEntityReference(entity_type=&#x27;samples&#x27;, entity_id=&#x27;385b44fb-4f25-4715-ae14-4c6147621e82&#x27;, catalog=&#x27;dcp55&#x27;), num_contributions=4, attempts=9), DocumentTally(entity=CataloguedEntityReference(entity_type=&#x27;samples&#x27;, entity_id=&#x27;60e5ce45-b58c-4cb3-8066-83c2cbb8caf9&#x27;, catalog=&#x27;dcp55&#x27;), num_contributions=4, attempts=9)] Traceback (most recent call last):   File &quot;&#x2f;var&#x2f;task&#x2f;azul&#x2f;indexer&#x2f;index_controller.py&quot;, line 280, in aggregate     self.index_queue_service.aggregate(tallies, retry=retry)     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^   File &quot;&#x2f;var&#x2f;task&#x2f;azul&#x2f;indexer&#x2f;index_queue_service.py&quot;, line 290, in aggregate     self.index_service.aggregate(tally_by_entity)     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^   File &quot;&#x2f;var&#x2f;task&#x2f;azul&#x2f;indexer&#x2f;index_service.py&quot;, line 489, in aggregate     assert len(aggregate.sources) == 1, R(            ^^^^^^^^^^^^^^^^^^^^^^^^^^^ AssertionError: R(&#x27;Entity has an invalid number of sources&#x27;, CataloguedEntityReference(entity_type=&#x27;samples&#x27;, entity_id=&#x27;385b44fb-4f25-4715-ae14-4c6147621e82&#x27;, catalog=&#x27;dcp55&#x27;), {TDRSourceRef(id=&#x27;1a449990-5cc7-4623-83a3-80440a0ab0bd&#x27;, spec=TDRSourceSpec(type=&lt;Type.bigquery: &#x27;bigquery&#x27;&gt;, domain=&lt;Domain.gcp: &#x27;gcp&#x27;&gt;, subdomain=&#x27;datarepo-03f25c54&#x27;, name=&#x27;hca_prod_4bcc16b57a4745bbb9c0be9d5336df2d__20240327_dcp2_20251104_dcp55&#x27;), prefix=Prefix(common=&#x27;&#x27;, partition=0)), TDRSourceRef(id=&#x27;ce551a61-822f-4786-b878-c62576b0e006&#x27;, spec=TDRSourceSpec(type=&lt;Type.bigquery: &#x27;bigquery&#x27;&gt;, domain=&lt;Domain.gcp: &#x27;gcp&#x27;&gt;, subdomain=&#x27;datarepo-5cffc4da&#x27;, name=&#x27;hca_prod_2079bb2e676e4bbf8c68f9c6459edcbb__20240327_dcp2_20251104_dcp55&#x27;), prefix=Prefix(common=&#x27;&#x27;, partition=0))}) |
---

```
[WARNING]	2025-11-15T11:12:17.211Z	e2f59684-9ad3-5ab3-8301-f716f8ec3754	azul.indexer.index_controller	Failed to aggregate tallies: [DocumentTally(entity=CataloguedEntityReference(entity_type='samples',
 entity_id='1eb02140-6c37-493c-9dcc-c2f5fb8e4ae0', catalog='dcp55')
, num_contributions=4, attempts=9), DocumentTally(entity=CataloguedEntityReference(entity_type='samples', entity_id='c494198d-97df-42d7-9656-5bf9c2a04ac8', catalog='dcp55'),
 num_contributions=4, attempts=9)]
Traceback (most recent call last):
  File ""/var/task/azul/indexer/index_controller.py"", line 280, in aggregate
    self.index_queue_service.aggregate(tallies, retry=retry)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
  File ""/var/task/azul/indexer/index_queue_service.py"", line 290, in aggregate
    self.index_service.aggregate(tally_by_entity)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File ""/var/task/azul/indexer/index_service.py"", line 489, in aggregate
    assert len(aggregate.sources) == 1, R(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: R('Entity has an invalid number of sources', CataloguedEntityReference(entity_type='samples',
 entity_id='1eb02140-6c37-493c-9dcc-c2f5fb8e4ae0', catalog='dcp55'),
 {TDRSourceRef(id='0d69b7d1-bc70-4bd3-946a-db6d49042778',
 spec=TDRSourceSpec(type=<Type.bigquery: 'bigquery'>, domain=<Domain.gcp: 'gcp'>,
 subdomain='datarepo-b42d0841', name='hca_prod_9c20a245f2c043ae82c92232ec6b594f__20220212_dcp2_20251104_dcp55'),
 prefix=Prefix(common='', partition=0)), TDRSourceRef(id='ce551a61-822f-4786-b878-c62576b0e006',
 spec=TDRSourceSpec(type=<Type.bigquery: 'bigquery'>, domain=<Domain.gcp: 'gcp'>, 
subdomain='datarepo-5cffc4da', name='hca_prod_2079bb2e676e4bbf8c68f9c6459edcbb__20240327_dcp2_20251104_dcp55'), prefix=Prefix(common='', partition=0))})

[WARNING]	2025-11-15T11:20:09.249Z	969edacd-7918-507f-82c1-bb5f1d00687a	azul.indexer.index_controller	Failed to aggregate tallies: [DocumentTally(entity=CataloguedEntityReference(entity_type='samples', entity_id='a75d5f39-acab-4b8f-88f2-64c7bdaa6752', catalog='dcp55'), num_contributions=13, attempts=9), DocumentTally(entity=CataloguedEntityReference(entity_type='samples', entity_id='d276045f-d040-4f29-b316-167429bb6c16', catalog='dcp55'), num_contributions=9, attempts=9)]
Traceback (most recent call last):
  File ""/var/task/azul/indexer/index_controller.py"", line 280, in aggregate
    self.index_queue_service.aggregate(tallies, retry=retry)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
  File ""/var/task/azul/indexer/index_queue_service.py"", line 290, in aggregate
    self.index_service.aggregate(tally_by_entity)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File ""/var/task/azul/indexer/index_service.py"", line 489, in aggregate
    assert len(aggregate.sources) == 1, R(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: R('Entity has an invalid number of sources', CataloguedEntityReference(entity_type='samples', entity_id='a75d5f39-acab-4b8f-88f2-64c7bdaa6752', catalog='dcp55'), {TDRSourceRef(id='ce551a61-822f-4786-b878-c62576b0e006', spec=TDRSourceSpec(type=<Type.bigquery: 'bigquery'>, domain=<Domain.gcp: 'gcp'>, subdomain='datarepo-5cffc4da', name='hca_prod_2079bb2e676e4bbf8c68f9c6459edcbb__20240327_dcp2_20251104_dcp55'), prefix=Prefix(common='', partition=0)), TDRSourceRef(id='1a449990-5cc7-4623-83a3-80440a0ab0bd', spec=TDRSourceSpec(type=<Type.bigquery: 'bigquery'>, domain=<Domain.gcp: 'gcp'>, subdomain='datarepo-03f25c54', name='hca_prod_4bcc16b57a4745bbb9c0be9d5336df2d__20240327_dcp2_20251104_dcp55'), prefix=Prefix(common='', partition=0))})"

[WARNING]	2025-11-15T13:13:16.454Z	3490c834-a9de-537f-80aa-a9a6d7459ff5	azul.indexer.index_controller	Failed to aggregate tallies: [DocumentTally(entity=CataloguedEntityReference(entity_type='samples', entity_id='385b44fb-4f25-4715-ae14-4c6147621e82', catalog='dcp55'), num_contributions=4, attempts=9), DocumentTally(entity=CataloguedEntityReference(entity_type='samples', entity_id='60e5ce45-b58c-4cb3-8066-83c2cbb8caf9', catalog='dcp55'), num_contributions=4, attempts=9)]
Traceback (most recent call last):
  File ""/var/task/azul/indexer/index_controller.py"", line 280, in aggregate
    self.index_queue_service.aggregate(tallies, retry=retry)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
  File ""/var/task/azul/indexer/index_queue_service.py"", line 290, in aggregate
    self.index_service.aggregate(tally_by_entity)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File ""/var/task/azul/indexer/index_service.py"", line 489, in aggregate
    assert len(aggregate.sources) == 1, R(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: R('Entity has an invalid number of sources', CataloguedEntityReference(entity_type='samples', entity_id='385b44fb-4f25-4715-ae14-4c6147621e82', catalog='dcp55'), {TDRSourceRef(id='1a449990-5cc7-4623-83a3-80440a0ab0bd', spec=TDRSourceSpec(type=<Type.bigquery: 'bigquery'>, domain=<Domain.gcp: 'gcp'>, subdomain='datarepo-03f25c54', name='hca_prod_4bcc16b57a4745bbb9c0be9d5336df2d__20240327_dcp2_20251104_dcp55'), prefix=Prefix(common='', partition=0)), TDRSourceRef(id='ce551a61-822f-4786-b878-c62576b0e006', spec=TDRSourceSpec(type=<Type.bigquery: 'bigquery'>, domain=<Domain.gcp: 'gcp'>, subdomain='datarepo-5cffc4da', name='hca_prod_2079bb2e676e4bbf8c68f9c6459edcbb__20240327_dcp2_20251104_dcp55'), prefix=Prefix(common='', partition=0))})
```

--- a/deployments/prod/environment.py
+++ b/deployments/prod/environment.py
@@ -1745,10 +1745,7 @@ dcp54_sources = mkdict(dcp53_sources, 521, mkdelta([
     mksrc('bigquery', 'datarepo-dcd21441', 'hca_prod_f77290ae0d7b4239b0fe3cf2c9e8858d__20250203_dcp2_20250930_dcp54'),
 ]))
 
-dcp55_sources = mkdict(dcp54_sources, 526, mkdelta([
-    mksrc('bigquery', 'datarepo-5cffc4da', 'hca_prod_2079bb2e676e4bbf8c68f9c6459edcbb__20240327_dcp2_20251104_dcp55'),
-    mksrc('bigquery', 'datarepo-03f25c54', 'hca_prod_4bcc16b57a4745bbb9c0be9d5336df2d__20240327_dcp2_20251104_dcp55'),
-    mksrc('bigquery', 'datarepo-b42d0841', 'hca_prod_9c20a245f2c043ae82c92232ec6b594f__20220212_dcp2_20251104_dcp55'),
+dcp55_sources = mkdict(dcp54_sources, 523, mkdelta([
     mksrc('bigquery', 'datarepo-d840670b', 'hca_prod_b10cd3143e7144379a1677028d243e81__20251104_dcp2_20251104_dcp55'),
     mksrc('bigquery', 'datarepo-bb0a7bc5', 'hca_prod_c7a342eb777e447995ce468f5bbcd893__20251104_dcp2_20251104_dcp55'),
 ]))

